Sources/Differ/Diff+AppKit.swift

```diff
@@ -10,38 +10,92 @@ extension NSTableView {
     ///   - newData:            Data which reflects the current state of `NSTableView`
     ///   - deletionAnimation:  Animation type for deletions
     ///   - insertionAnimation: Animation type for insertions
-    ///   - indexPathTransform: Closure which transforms zero-based `IndexPath` to desired `IndexPath`
+    ///   - rowIndexTransform:  Closure which transforms a zero-based row to the desired index
     public func animateRowChanges<T: Collection>(
         oldData: T,
         newData: T,
         deletionAnimation: NSTableView.AnimationOptions = [],
         insertionAnimation: NSTableView.AnimationOptions = [],
-        indexPathTransform: (IndexPath) -> IndexPath = { $0 }
+        rowIndexTransform: (Int) -> Int = { $0 }
     ) where T.Element: Equatable {
-        apply(
-            oldData.extendedDiff(newData),
-            deletionAnimation: deletionAnimation,
-            insertionAnimation: insertionAnimation,
-            indexPathTransform: indexPathTransform
-        )
+        let patch = extendedPatch(from: oldData, to: newData)
+        apply(patch, deletionAnimation: deletionAnimation, insertionAnimation: insertionAnimation, rowIndexTransform: rowIndexTransform)
     }
 
-    /// Animates rows which changed between oldData and newData.
+    /// Animates rows which changed between oldData and newData using custom `isEqual`.
     ///
     /// - Parameters:
     ///   - oldData:            Data which reflects the previous state of `NSTableView`
     ///   - newData:            Data which reflects the current state of `NSTableView`
     ///   - isEqual:            A function comparing two elements of `T`
     ///   - deletionAnimation:  Animation type for deletions
     ///   - insertionAnimation: Animation type for insertions
-    ///   - indexPathTransform: Closure which transforms zero-based `IndexPath` to desired `IndexPath`
+    ///   - rowIndexTransform:  Closure which transforms a zero-based row to the desired index
+    public func animateRowChanges<T: Collection>(
+        oldData: T,
+        newData: T,
+        isEqual: EqualityChecker<T>,
+        deletionAnimation: NSTableView.AnimationOptions = [],
+        insertionAnimation: NSTableView.AnimationOptions = [],
+        rowIndexTransform: (Int) -> Int = { $0 }
+    ) {
+        withoutActuallyEscaping(isEqual) { isEqual in
+            func wrap(_ wrapped: T.Element) -> CustomEquatable<T> { CustomEquatable(isEqual: isEqual, wrapped: wrapped) }
+
+            animateRowChanges(oldData: oldData.map(wrap), newData: newData.map(wrap), deletionAnimation: deletionAnimation, insertionAnimation: insertionAnimation, rowIndexTransform: rowIndexTransform)
+        }
```

Comment:
I don't like calling `animateRowChanges(oldData: newData:deletionAnimation:insertionAnimation:rowIndexTransform:)` from here, as it forces us to create this `Equatable` wrapper and work with `withoutActuallyEscaping(_:)`.

Can we maybe turn this around, i.e. compute the patch based on `isEqual` and apply it in here, and call this method with `isEqual: ==` from `animateRowChanges(oldData: newData:deletionAnimation:insertionAnimation:rowIndexTransform:)`? This would require adding an `isEqual:` parameter to the patch-family of methods, which might be helpful to other applications as well.

Reply:
Yes I'd also prefer that, however I at first didn't see a `extendedPatch` function/init based that allowed to specify the `isEqual` to be used. Do you have suggestion what to use for this? Would `extendedDiff(…isEqual:…).patch(…)` be the right thing?
(I was a bit hesitant to introduce a new function or parameter to `extendedPatch`, as it seemed to reach further into the codebase rather than just fixing the `NSTableView` adapter)

Reply:
I believe `extendedDiff(_:isEqual:).patch(from:to:sort:)` would be the right thing, though I'm a little confused as to why `patch(from:to:sort:)` doesn't require the comparator as well.